Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM golang:1.12-alpine as builder
+
+WORKDIR /go/src/github.com/cloudflare/cloudflared/
+COPY . .
+
+RUN apk add --no-cache gcc musl-dev git upx ca-certificates
```

Comment:
Should we run `apk update && apk add`? I usually see that done in Dockerfiles, to pick up changes to the registry between the base image build and the Dockerfile build.

Reply:
[Upstream](https://github.com/docker-library/golang/blob/d6715f0733f5282836cad2867a0a2b107fc35147/1.12/alpine3.9/Dockerfile) doesn't seem to do it and in general I'm not a fan of doing so because it means we're sliding the updates between builds. This should mean we layer our container properly.

Reply:
Since this is somewhat subjective I'll leave it for you to decide what you prefer.